xchange-exmo/src/main/java/org/knowm/xchange/exmo/service/ExmoAdapters.java

```diff
@@ -52,14 +55,32 @@ public static Balance adaptBalance(
   }
 
   public static Ticker adaptTicker(CurrencyPair currencyPair, Map<String, String> data) {
+
+    final BigDecimal last = new BigDecimal(data.get("last_trade"));
+    final BigDecimal buyPrice = new BigDecimal(data.get("buy_price"));
+
+    BigDecimal priceChange = null;
+    BigDecimal priceChangePercent = null;
+
+    if (ExmoMarketDataHolder.getCloseBuyPrice(currencyPair) != null) {
```

Comment:
this assumes that some one is adapting ticker in a specified order and will fail in multi threaded applications or any other use case that does not respect this hidden requirement

Reply:
Agree with you, but there is no way to calc priceChange and priceChangePercent without holding data. And without that it works correctly only without this general ticker data fields

Reply:
you should calculate this in your application code on a given list, NOT in a builder of a single ticker, the way you are implementing this is unacceptable and prone to errors